### DISCUSSION TITLE
mpiexec not working

### DISCUSSION THREAD

**Discussion by DengMichael** [wenceorg/comp4187#40] [Q&A] [answered]
Hi, I wanted to test running the following:

```
sudo mpiexec -n 4 tests/test_one_dim.py -v
```

But I've been getting the following:
```
[proxy:0:0@WINDELL-ROAVSEC] HYDU_create_process (utils/launch/launch.c:74): execvp error on file tests/test_one_dim.py (Permission denied)
[proxy:0:0@WINDELL-ROAVSEC] HYDU_create_process (utils/launch/launch.c:74): execvp error on file tests/test_one_dim.py (Permission denied)
[proxy:0:0@WINDELL-ROAVSEC] HYDU_create_process (utils/launch/launch.c:74): execvp error on file tests/test_one_dim.py (Permission denied)
[proxy:0:0@WINDELL-ROAVSEC] HYDU_create_process (utils/launch/launch.c:74): execvp error on file tests/test_one_dim.py (Permission denied)
```

I've apt installed all the packages on the MPI Administrivia page to no avail.
Any suggestions would be appreciated.

**Comment by annereinarz** [comment, marked as answer]
A few points to start:
- why are you using sudo to run this?
- why are you running the tests individually as opposed to using pytest?
The error you have suggests that the permissions on the file are not set correctly. Running "ls -l" will show you the permissions, which may clarify what is going wrong.

**Reply by DengMichael** [reply]
- I wanted to try it with sudo after I tried it without sudo. (Sorry I probably should've clarified!)
- I wanted to understand the point structure better in parallel by inspecting values when run in parallel

It was indeed a permissions issue, I thought I had changed them before but apparently I did it for the wrong copy, thanks for the clarification!

**Reply by wence-** [reply]
FWIW, if you want to explicitly run the test in parallel you can do:

```
mpiexec -n 4 python -m pytest tests/test_one_dim.py
```

If you just use python to run the file no tests are executed because pytest won't be involved.
